internal/cache/cache.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+)
+
+// NOTE: this is heavily based on patrickmn/go-cache:
+// https://github.com/patrickmn/go-cache
```

Comment:
Given "heavily" equals to "substantial", we need to include the full MIT license notice to comply. https://github.com/patrickmn/go-cache/blob/master/LICENSE#L10-L11